src/main/java/org/jboss/staxmapper/XMLExtendedStreamReaderImpl.java

```diff
@@ -451,9 +451,9 @@ public long getLongAttributeValue(final int index) throws XMLStreamException {
                 throw intParseException(e, getLocation());
             }
         } else if (kind.isEnum()) {
-            final List<T> list = new ArrayList<T>();
+            final List<T> list = new ArrayList<>();
             for (String s : Spliterable.over(getAttributeValue(index), ' ')) {
-                list.add(kind.cast(Enum.valueOf(kind.asSubclass(Enum.class), s)));
+                list.add((T) kind.cast(Enum.valueOf(kind.asSubclass(Enum.class), s)));
```

Comment:
Still it's better to fix it properly.  Maybe assign the problematic inner expression to a local variable to "help" the inference.